linera-execution/src/runtime.rs

```diff
@@ -1590,6 +1590,27 @@ impl ContractRuntime for ContractSyncRuntimeHandle {
         Ok(app_id)
     }
 
+    fn create_data_blob(&mut self, bytes: Vec<u8>) -> Result<BlobId, ExecutionError> {
+        let blob = Blob::new_data(bytes);
+        let blob_id = blob.id();
+        self.inner().transaction_tracker.add_created_blob(blob);
+        Ok(blob_id)
+    }
+
+    fn publish_module(
+        &mut self,
+        contract: Bytecode,
+        service: Bytecode,
+        vm_runtime: VmRuntime,
+    ) -> Result<ModuleId, ExecutionError> {
+        let (blobs, module_id) =
+            crate::runtime::create_bytecode_blobs_sync(contract, service, vm_runtime);
+        for blob in blobs {
+            self.inner().transaction_tracker.add_created_blob(blob);
+        }
```

Comment:
Remember to charge the fees.

Reply:
Good question.
I am not sure we need, but if so, then it also applies to `open_chain` and `create_application`.
So, let me check.

Reply:
My conclusion is that the tracking is done by one single function, which is `track_blob_published`.
That function is called from the `process_txn_outcome` in `linera-core`.
Therefore, I believe that the blobs created here are processed in the same manner as other blobs generated in the block, and no additional processing code is required.